daemon/cmd/endpoint.go

```diff
@@ -763,7 +763,7 @@ func (d *Daemon) EndpointUpdate(id string, cfg *models.EndpointConfigurationSpec
 			return api.Error(PatchEndpointIDConfigFailedCode, err)
 		}
 	}
-	if err := ep.UpdateReferences(d.endpointManager); err != nil {
+	if err := d.endpointManager.UpdateReferences(ep); err != nil {
```

Comment:
Incidentally, I noticed that this path is actually broken. If we intend to change any of the Identifiers from this path, then this code is not actually removing the previous identifiers, it's only creating new references in the endpointmanager for the endpoint. 

I don't think this is actually a viable use of the config API today so I didn't bother to fix it, and also I was trying to keep the scope of this PR relatively small so I can move on to the PR that I want to build on top of these changes.